modules/rpc/internal/stats/metrics.go

```diff
@@ -33,27 +33,47 @@ const (
 	TagMiddleware = "middleware"
 )
 
-// HTTPTags creates metrics scope with defined tags
 var (
-
-	// RPCTags creates metrics scope with defined tags
-	RPCTags = map[string]string{
+	rpcTags = map[string]string{
 		TagModule: "rpc",
 		TagType:   "request",
 	}
+)
 
+// Client is a client for stats.
+type Client interface {
 	// RPCAuthFailCounter counts auth failures
-	RPCAuthFailCounter tally.Counter
+	RPCAuthFailCounter() tally.Counter
 	// RPCHandleTimer is a turnaround time for rpc handler
-	RPCHandleTimer tally.Scope
+	RPCHandleTimer() tally.Scope
 	// RPCPanicCounter counts panics occurred for rpc handler
-	RPCPanicCounter tally.Counter
-)
+	RPCPanicCounter() tally.Counter
+}
+
+// NewClient returns a new Client for the given tally.Scope.
+func NewClient(scope tally.Scope) Client {
+	rpcTagsScope := scope.Tagged(rpcTags)
+	return &client{
+		rpcTagsScope.Tagged(map[string]string{TagMiddleware: "auth"}).Counter("fail"),
```

Comment:
I would love to talk about this - I leave out the field names on purpose. This is so that if a new field is added or a field is deleted later, I want there to be a compile-time error here. By adding the tags, I've had a lot more bugs.